feat: @llamaindex/chat-ui components (#1)
run-llama/chat-ui: README.md

## Usage

1. Install the package

```sh
npm install @llamaindex/chat-ui
```

2. Configure your `tailwind.config.ts` to include the chat-ui components

```ts
module.exports = {
  content: [
    'app/**/*.{ts,tsx}',
    'node_modules/@llamaindex/chat-ui/**/*.{ts,tsx}',
  ],
  // ...
}
```

3. Import the components and use them

```tsx
import { ChatSection } from '@llamaindex/chat-ui'
import { useChat } from 'ai/react'

const ChatExample = () => {
  const handler = useChat()
  return <ChatSection handler={handler} />
}
```

## Custom theme

> You can customize the theme by overriding the default styles.

Reviewer: ```suggestion
```

For a list of all available CSS variables, please refer to the [Shadcn Theme Config](https://ui.shadcn.com/themes).

## Component Composition

Components are designed to be composable. You can use them as is, or extend them with your own styles and behaviors.

You can use just need to use ChatSection like this:

```tsx
import '@llamaindex/chat-ui/styles.css'
import { ChatSection, ChatMessages, ChatInput } from '@llamaindex/chat-ui'
import { useChat } from 'ai/react' // or other chat context

const ChatExample = () => {
  const handler = useChat()
  return <ChatSection handler={handler} />
}
```

Or you can extend them with your own children components and styles:

```tsx

> import '@llamaindex/chat-ui/styles.css'

Reviewer: ```suggestion
```

import { ChatSection, ChatMessages, ChatInput } from '@llamaindex/chat-ui'
import LlamaCloudSelector from './components/LlamaCloudSelector' // your custom component
import { useChat } from 'ai/react'

const ChatExample = () => {

>   const handler = useChat()
>   return (
>     <ChatSection handler={handler}>
>       <ChatMessages />
>       <ChatInput>
>         <ChatInput.Preview />
>         <ChatInput.Form className="bg-lime-500"> {/* custom styles */}
>           <ChatInput.Field type="textarea" />
>           <ChatInput.Upload />
>           <LlamaCloudSelector /> {/* custom component */}
>           <ChatInput.Submit />
>         </ChatInput.Form>
>       </ChatInput>
>     </ChatSection>
>   )

Reviewer: i guess the example would be easier like this?
```suggestion
  const [requestData, setRequestData] = useState<any>();
  const handler = useChat()
  return (
    <ChatSection handler={handler}>
      <ChatMessages />
      <ChatInput 
        setRequestData={setRequestData} >
        <ChatInput.Preview />
        <ChatInput.Form className="bg-lime-500"> {/* custom styles */}
          <ChatInput.Field type="textarea" />
          <ChatInput.Upload />
          <LlamaCloudSelector requestData={requestData} /> {/* custom component */}
          <ChatInput.Submit />
        </ChatInput.Form>
      </ChatInput>
    </ChatSection>
  )
```

}
```

> Your custom component can use provided hooks like `useChat` to access the chat context.

Reviewer: needs to be updated after the discussion above

```tsx
import { useChatInput } from '@llamaindex/chat-ui'

const LlamaCloudSelector = () => {

>   const { data, setData } = useChat()

Reviewer: update with changes of example above

Components are designed to be composable. You can use them as is, or extend them with your own styles and behaviors.

So the easiest way to get started is to connect the whole `ChatSection` component with [vercel/ai](https://github.com/vercel/ai) like this:

```tsx
import { ChatSection } from '@llamaindex/chat-ui'
import { useChat } from 'ai/react'

const ChatExample = () => {
  const handler = useChat()
  return <ChatSection handler={handler} />
}
```

Or you can extend them with your own children components and styles:

```tsx
import { ChatSection, ChatMessages, ChatInput } from '@llamaindex/chat-ui'
import LlamaCloudSelector from './components/LlamaCloudSelector' // your custom component
import { useChat } from 'ai/react'

const ChatExample = () => {
  const handler = useChat()
  return (
    <ChatSection handler={handler}>
      <ChatMessages />
      <ChatInput>
        <ChatInput.Preview />
        <ChatInput.Form className="bg-lime-500">
          <ChatInput.Field type="textarea" />
          <ChatInput.Upload />
          <LlamaCloudSelector /> {/* custom component */}
          <ChatInput.Submit />
        </ChatInput.Form>
      </ChatInput>
    </ChatSection>
  )
}
```

> Your custom component can use provided hooks like `useChatUI` to access the chat context.

Reviewer: ```suggestion
Your custom component can use the `useChatUI` hook to send additional data to the chat API endpoint:
```
